src/javascript/_common/third_party_links.js

```diff
@@ -11,6 +11,29 @@ const ThirdPartyLinks = (() => {
                 document.body.addEventListener('click', clickHandler);
             }
         });
+        document.body.addEventListener('click', checkTelegram);
+    };
+
+    const checkTelegram = (e) => {
+        if (!e.target) return;
+        const el_link = e.target.closest('a');
+        if (!el_link) return;
+
+        const dialog = document.querySelector('#telegram');
+        if (dialog && dialog.contains(el_link)) return;
+
+        if (isTelegramLink(el_link.href)) {
+            e.preventDefault();
+            // show a popup to remind clients to have Telegram app installed on their device
+            Dialog.confirm({
+                id               : 'telegram',
+                localized_message: localize(['Please ensure that you have the Telegram app installed on your device.', 'Click OK to proceed.']),
+            }).then((should_proceed) => {
+                if (should_proceed) {
+                    openThirdPartyLink(el_link.href);
+                }
+            });
+        }
```

Comment:
this has many things in common with clickHandler. perhaps we can refactor them to use a similar function with a few passed in variables? i.e. keep `checkTelegram` and `clickHandler` functions so our event listeners can call a function without passing variables, but inside them call another function that checks the link and handles showing the popup.